src/main/scala/is/hail/linalg/BlockMatrix.scala

```diff
@@ -1100,22 +1098,23 @@ class WriteBlocksRDD(path: String,
     val blockRow = split.index
     val nRowsInBlock = gp.blockRowNRows(blockRow)
 
-    val dosPerBlockCol = Array.tabulate(gp.nBlockCols) { blockCol =>
+    val outPerBlockCol = Array.tabulate(gp.nBlockCols) { blockCol =>
       val nColsInBlock = gp.blockColNCols(blockCol)
 
       val i = gp.coordinatesBlock(blockRow, blockCol)
       val filename = path + "/parts/" + partFile(d, i)
 
-      val dos = new DataOutputStream(sHadoopBc.value.value.unsafeWriter(filename))
-      dos.writeInt(nRowsInBlock)
-      dos.writeInt(nColsInBlock)
-      dos.writeBoolean(true) // transposed, stored row major
-
-      dos
+      val os = sHadoopBc.value.value.unsafeWriter(filename)
+      val out = BlockMatrix.bufferSpec.buildOutputBuffer(os)
+      
+      out.writeInt(nRowsInBlock)
+      out.writeInt(nColsInBlock)
+      out.writeBoolean(true) // transposed, stored row major
+      out.flush() // align block row with buffer
```

Comment:
Yeah, I'm against this.  I don't think it buys you anything but makes the code more brittle.